src/components/MultiHopTrade/components/LimitOrder/components/LimitOrderInput.tsx

```diff
@@ -172,6 +182,18 @@ export const LimitOrderInput = ({
     [handleFormSubmit],
   )
 
+  const assetFilterPredicate = useCallback((asset: Asset) => {
+    return (
+      asset.chainId !== KnownChainIds.SolanaMainnet &&
+      isEvmChainId(asset.chainId) &&
+      !isNativeEvmAsset(asset.assetId)
+    )
+  }, [])
+
+  const chainIdFilterPredicate = useCallback((chainId: ChainId) => {
+    return chainId !== KnownChainIds.SolanaMainnet && isEvmChainId(chainId)
```

Comment:
preferably-blocking: right-side should be enough here

Reply:
Here ya go 3fb23025c7c6cfaf7382ef1d9baff820ecafc723